maven-spy/src/main/java/org/jenkinsci/plugins/pipeline/maven/eventspy/handler/DependencyResolutionResultHandler.java

```diff
@@ -42,13 +47,34 @@ protected boolean _handle(DependencyResolutionResult result) {
         Xpp3Dom root = new Xpp3Dom("DependencyResolutionResult");
         root.setAttribute("class", result.getClass().getName());
 
-        /*
         Xpp3Dom dependenciesElt = new Xpp3Dom("resolvedDependencies");
         root.addChild(dependenciesElt);
-        // don't cast to Dependency to not have a classloading issue at runtime
-        for (Object dependency: result.getResolvedDependencies()) {
-            dependenciesElt.addChild(newElement("dependency", dependency.toString()));
-        }*/
+
+        for (Dependency dependency: result.getResolvedDependencies()) {
```

Comment:
I haven't seen an issue thus far. Our build slaves use either 3.2.5 or 3.3.9.

Is is possible another plugin is adding the dependency to the class path?